### DISCUSSION TITLE
What determines which package version gets hoisted to `.pnpm/node_modules`?

### DISCUSSION THREAD

**Discussion by Torsten85** [pnpm/pnpm#5300] [Q&A]
Hello,

I've a pnpm monorepo with two packages, both packages have a dependency on `@types/react` but one on `@^17` the other on `@^18`. 
I know by default, all packages gets hoisted ( `hoist_pattern=['*']` )
But currently I see version 17 in `node_modules/.pnpm/node_modules/@types/react`. Is this configurable or what determines which version gets picked? 

Thanks in advance

**Comment by zkochan** [comment]
pnpm just walks the dependency graph and picks the first one it finds by a given unique name. If you need proper hoisting, your only option is to use `node-linker=hoisted`.

**Reply by tianleiyiji123** [reply]
> 

but, Because there are two types of react and both are hoisted, then react gets a wrong version types of react,how can id do ?

**Reply by zkochan** [reply]
Then try to disable hoisting. `hoist=false`.
